compiler/src/tests/cxxfrontend.cc

```diff
@@ -234,6 +234,7 @@ void test_visitors()
     ClassDeclaration *cd = ClassDeclaration::create(loc, Identifier::idPool("TypeInfo"), NULL, NULL, true);
     assert(cd->isClassDeclaration() == cd);
     assert(cd->vtblOffset() == 1);
```

Comment:
this assert failing means that 
```d
if (isCOMinterface() || isCPPinterface())
    return 0;
return 1;
```
is returning the wrong value. 
Both of those methods should return `false` for `class`es (i.e. only for interfaces should they possibly return `true`). I'm not really sure why that is happening. You might want to check what the actual return value is. If it is `0` or if it is some other value (which would indicate a corrupt vtable on the C++ side).

Reply:
It is odd that is only happens with G++/MSVC(???) as the C++ compiler

Reply:
okay i'd look at the actual return value

Reply:
@thewilsonator I'm a bit lost here, how do I figure that out??